url-updater/src/main/java/com/devonfw/tools/ide/url/folderhandling/UrlChecksum.java

```diff
@@ -0,0 +1,129 @@
+package com.devonfw.tools.ide.url.folderhandling;
+
+import com.devonfw.tools.ide.url.folderhandling.abstractUrlClasses.AbstractUrlFile;
+
+import java.io.*;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+public class UrlChecksum extends AbstractUrlFile {
+
+    final HttpClient client = HttpClient.newBuilder().build();
```

Comment:
should be `private`.
IMHO we should keep logic like `HttpClient` out of the "model" (I would keep it in `AbstractCrawler` that already has an `HttpClient` instance).

Reply:
Done